graviti/schema/builtin.py

```diff
@@ -258,6 +310,18 @@ def __getitem__(self, index: Union[int, str, slice]) -> Union[PortexType, List[P
 
         return self.fields[index].type
 
+    def _pyarrow_type_factory(self) -> "paDataType":
+        return pa.struct(self.fields._to_pyarrow())  # pylint: disable=[no-member, protected-access]
```

Comment:
will it be better if we use a flag rather than a private method `to_pyarrow`?
flag is helpful for the `protected-access` issue